### PR TITLE
[12.0][MIG] rma_sale

### PR DESCRIPTION
**Pull request by grindtildeath**

### CODE REVIEW

**Comment by max3903** [comment]
@osi-scampbell, @murtuzasaleh Can you review, test and put your comments here?

**Comment by murtuzasaleh** [comment]
@grindtildeath While testing in the local database I am facing below issues.

Issue 1: 'bool' object has no attribute 'split'

Steps to reproduce:- 
 
1)  Create New database and install `rma_sale` module and Go to Inventory >> Customer RMA >> Customer RMA Group, Create Customer RMA Group then click on "Add From Sale Order" button and clicked to add a line then an error has occurred.

Below is the screenshot of it.
issue_1

Issue 2: creation/update: a mandatory field is not correctly set.  an object with reference uom_id

Steps to reproduce:- 

Go to Inventory >> Customer RMA >> Customer RMA Group, Click on "Add From Invoice" button >> Select Invoices lines and click on Confirm Button then an error has occurred.

Below is the screenshot of it.

issue_2

Issue 3: key error `picking_id`

Steps to reproduce:- 

Go to Inventory >> Customer RMA >> Customer RMA Group, Click on "Add From Stock Move" button >> add a line and click on Confirm Button then an error has occurred.

issue_3

File: rma_sale/wizards/rma_order_line_make_sale_order_view.xml
Comment (murtuzasaleh) [inline rma_sale/wizards/rma_order_line_make_sale_order_view.xml]: Please Use `odoo` instead of `openerp` and also remove 'data' in all .xml file if no need.

**Review by murtuzasaleh — CHANGES_REQUESTED** [review]

**Comment by grindtildeath** [comment]
@murtuzasaleh Thanks for your review.
I added b65f6b6 here to fix XML files and opened PR #71 to fix your Issue 3.
However I could not reproduce your Issue 1 and Issue 2.
For Issue 1, from what I can read in your screenshot, it seems the sale.order.line doesn't have a name, maybe because it wasn't saved?
For Issue 2, couldn't this be related to group/rights settings on UOM ? I tried with and without the group "Manage Multiple Units of Measure" but didn't get error.

**Comment by murtuzasaleh** [comment]
> However I could not reproduce your Issue 1 and Issue 2.
> For Issue 1, from what I can read in your screenshot, it seems the sale.order.line doesn't have a name, maybe because it wasn't saved?

Follow the below steps:-

1) First **Create New database** and install a rma_sale module and Go to Inventory >> Customer RMA >> Customer RMA Group, Create Customer RMA Group then click on "Add From Sale Order" button and clicked to add a line then an error has occurred.

```
Traceback (most recent call last):
  File "/home/serpentcs/workspace/12.0/odoo/http.py", line 654, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/serpentcs/workspace/12.0/odoo/http.py", line 312, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/home/serpentcs/workspace/12.0/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/home/serpentcs/workspace/12.0/odoo/http.py", line 696, in dispatch
    result = self._call_function(**self.params)
  File "/home/serpentcs/workspace/12.0/odoo/http.py", line 344, in _call_function
    return checked_call(self.db, args, *kwargs)
  File "/home/serpentcs/workspace/12.0/odoo/service/model.py", line 97, in wrapper
    return f(dbname, args, *kwargs)
  File "/home/serpentcs/workspace/12.0/odoo/http.py", line 337, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/serpentcs/workspace/12.0/odoo/http.py", line 939, in _call_
    return self.method(*args, **kw)
  File "/home/serpentcs/workspace/12.0/odoo/http.py", line 517, in response_wrap
    response = f(*args, **kw)
  File "/home/serpentcs/workspace/12.0/addons/web/controllers/main.py", line 962, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/home/serpentcs/workspace/12.0/addons/web/controllers/main.py", line 954, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/serpentcs/workspace/12.0/odoo/api.py", line 749, in call_kw
    return _call_kw_multi(method, model, args, kwargs)
  File "/home/serpentcs/workspace/12.0/odoo/api.py", line 736, in _call_kw_multi
    result = method(recs, args, *kwargs)
  File "/home/serpentcs/workspace/12.0/odoo/models.py", line 5472, in onchange
    snapshot1 = Snapshot(record, nametree)
  File "/home/serpentcs/workspace/12.0/odoo/models.py", line 5379, in _init_
    if subnames else record[name]
  File "/home/serpentcs/workspace/12.0/odoo/models.py", line 5088, in _getitem_
    return self._fields[key].__get__(self, type(self))
  File "/home/serpentcs/workspace/12.0/odoo/fields.py", line 986, in _get_
    self.determine_draft_value(record)
  File "/home/serpentcs/workspace/12.0/odoo/fields.py", line 1106, in determine_draft_value
    self._compute_value(record)
  File "/home/serpentcs/workspace/12.0/odoo/fields.py", line 1042, in _compute_value
    getattr(records, self.compute)()
  File "/home/serpentcs/workspace/12.0/odoo/models.py", line 1570, in _compute_display_name
    names = dict(self.name_get())
  File "/home/serpentcs/workspace/FSM_MS/12.0-mig-rma_sale/rma_sale/models/sale_order_line.py", line 45, in name_get
    return super(SaleOrderLine, self).name_get()
  File "/home/serpentcs/workspace/12.0/addons/sale/models/sale.py", line 1526, in name_get
    name = '%s - %s' % (so_line.order_id.name, so_line.name.split('\n')[0] or so_line.product_id.name)
AttributeError: 'bool' object has no attribute 'split'
```



> For Issue 2, couldn't this be related to group/rights settings on UOM ? I tried with and without the group "Manage Multiple Units of Measure" but didn't get error.

Have you selected uom_id in a wizard? If yes then try it with without selecting uom_id and you got the error.

**Comment by osi-scampbell** [comment]
I could not replicate Issue 1 or Issue 2 on my system. Was able to add to an RMA group from stock move and sale order. We have the PR to fix add from invoice. One suggestion I had was possibly adding sale_management to the dependency list? Currently this module depends on rma and sale_stock, sale_stock depends on stock_account and sale. Since rma_purchase installs Purchasing I thought maybe we would want rma_sale to install Sales Management. Either way, after we get that other PR merged we can approve this one.

**Comment by max3903** [comment]
@aheficent @jbeficent #71 has been merged. We should be good to merge this one now. Thank you!

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Just fixed up last commit and rebased. Functional tests good so far. Will merge when ci finish.

**Comment by AaronHForgeFlow** [comment]
@murtuzasaleh I think your comments were attended here.

**Comment by murtuzasaleh** [comment]
I have tested it locally and it works properly.

**Review by murtuzasaleh — APPROVED** [review]

**Pull request merged**
